pysph/tools/binder.py

```diff
@@ -0,0 +1,208 @@
+"""Make a directory compatible with mybinder.org and ready for
+upload to a Github repo.
+"""
+
+from pysph.solver.utils import get_files, mkdir
+import os
+import collections
+from shutil import copy
+import nbformat as nbf
+import argparse
+import sys
+import re
+import glob
+
+
+def find_viewer_type(path):
+    '''
+    Finds the type of viewer to use in the jupyter notebook.
+    Parses the log file at the file path and searches for
+    'dim=d', where 'dD' is taken to be the viewer type.
+
+    example: if 'dim=2', then what is returned is '2D'
+    '''
+
+    log_file_path = os.path.abspath(path) + '/*.log'
+    regex = 'dim=(\d)'
+    # The above '\d' is marked as an invalid escape sequence in PEP 8.
+    # It is a valid regex sequence that matches to a digit.
+
+    match_list = []
+    with open(glob.glob(log_file_path)[0], 'r') as file:
+        for line in file:
+            for match in re.finditer(regex, line, re.S):
+                match_text = match.group()
+                match_list.append(match_text)
+                if len(match_list) > 0:
+                    break
+            if len(match_list) > 0:
+                break
+    return match_list[0][-1] + 'D'
+
+
+def make_notebook(path, sim_name, config_dict={}):
+    '''
+    Makes a jupyter notebook to view simulation results stored in
+    a given directory
+
+    path: the directory conatining the output files
+
+    sim_name: name of the simulation
+              ex. 'cavity_output'
+
+    config_dict: configuration dictionary for the notbeook viewer [dict]
+                 ex. {'fluid': {'frame': 20}}
+    '''
+
+    viewer_type = find_viewer_type(path)
+    cell1_src = [
+        "import os\n",
+        "from pysph.tools.ipy_viewer import Viewer" + viewer_type
+    ]
+    cell2_src = [
+        "cwd = os.getcwd()"
+    ]
+    cell3_src = [
+        "viewer = Viewer" + viewer_type + "(cwd)\n",
+        "viewer.interactive_plot(" + str(config_dict) + ")"
+    ]
+
+    nb = nbf.v4.new_notebook()
+
+    nb.cells = [
+        nbf.v4.new_code_cell(source=cell1_src),
+        nbf.v4.new_code_cell(source=cell2_src),
+        nbf.v4.new_code_cell(source=cell3_src)
+    ]
+
+    nbf.write(nb, path+'/'+sim_name+'.ipynb')
```

Comment:
Use `os.path.join` instead of doing the `'/'` additions.

Reply:
Fixed.